File: src/main/java/nl/chimpgamer/networkmanager/extensions/discordbot/tasks/TokenExpiryTask.java
package nl.chimpgamer.networkmanager.extensions.discordbot.tasks;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import nl.chimpgamer.networkmanager.common.utils.Methods;
import nl.chimpgamer.networkmanager.extensions.discordbot.DiscordBot;
import nl.chimpgamer.networkmanager.extensions.discordbot.configurations.Setting;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.DCMessage;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.JsonEmbedBuilder;
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.Utils;
import nl.chimpgamer.networkmanager.extensions.discordbot.api.models.Token;

@RequiredArgsConstructor
@Getter
public class TokenExpiryTask implements Runnable {
    private final DiscordBot discordBot;
    private final Token token;

    @Override
    public void run() {
        if (!this.getDiscordBot().getDiscordUserManager().getTokens().contains(this.getToken())) {
            return;
        }

        String msgStr = DCMessage.REGISTRATION_TOKEN_EXPIRED.getMessage()
                .replace("%command_prefix%", Setting.DISCORD_COMMAND_PREFIX.getAsString());
        if (Methods.isJsonValid(msgStr)) {
            JsonEmbedBuilder jsonEmbedBuilder = JsonEmbedBuilder.fromJson(msgStr);
            Utils.editMessage(getToken().getMessage(), jsonEmbedBuilder.build());
        } else {
            Utils.editMessage(getToken().getMessage(), msgStr);
        }

        this.getDiscordBot().getDiscordUserManager().getTokens().remove(getToken());
        this.getDiscordBot().getNetworkManager().debug("Token: " + getToken().getToken() + " has been removed!");
    }
}

File: src/main/kotlin/nl/chimpgamer/networkmanager/extensions/discordbot/tasks/TokenExpiryTask.kt
package nl.chimpgamer.networkmanager.extensions.discordbot.tasks

import nl.chimpgamer.networkmanager.extensions.discordbot.DiscordBot
import nl.chimpgamer.networkmanager.extensions.discordbot.api.models.Token
import nl.chimpgamer.networkmanager.extensions.discordbot.configurations.DCMessage
import nl.chimpgamer.networkmanager.extensions.discordbot.modals.JsonMessageEmbed
import nl.chimpgamer.networkmanager.extensions.discordbot.utils.Utils

class TokenExpiryTask(private val discordBot: DiscordBot, private val token: Token) : Runnable {

    override fun run() {
        if (!this.discordBot.discordUserManager.tokens.contains(this.token)) return


        val msgStr = discordBot.messages.getString(DCMessage.REGISTRATION_TOKEN_EXPIRED)

        if (Utils.isJsonValid(msgStr)) {
            val jsonMessageEmbed = JsonMessageEmbed.fromJson(msgStr)
            token.interaction.editOriginalEmbeds(jsonMessageEmbed.toMessageEmbed()).queue {
                this.discordBot.discordUserManager.tokens.remove(this.token)
                this.discordBot.networkManager.debug("Token: ${this.token.token} has been removed!")
            }
        } else {
            token.interaction.editOriginal(msgStr).queue {
                this.discordBot.discordUserManager.tokens.remove(this.token)
                this.discordBot.networkManager.debug("Token: ${this.token.token} has been removed!")
            }
        }
    }
}